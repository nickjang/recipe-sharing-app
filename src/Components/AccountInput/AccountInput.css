.account-input {
  border: 0;
}

.main-input {
  display: flex;
  flex-direction: column;
  font-family: var(--note-font);
  text-align: right;
}

.main-input input {
  border: 1px solid var(--color-black);
}

.account-input .input-label {
  margin: auto 4px auto 0px;
}

.account-input .hint {
  margin-left: 4px;
}

.account-input .validation-error {
  display: block;
  margin: auto;
  max-width: fit-content;
}

@media all and (min-width: 345px) {
  .main-input {
    flex-direction: row;
    justify-content: flex-end;
  }
}
