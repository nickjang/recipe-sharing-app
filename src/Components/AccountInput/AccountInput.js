import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ValidationError from '../ValidationError/ValidationError';
import './AccountInput.css';

class AccountInput extends Component {
  render() {
    let inputLabel = this.props.id.replace(/-|_/g, ' ');
    inputLabel = inputLabel.charAt(0).toUpperCase() + inputLabel.slice(1) + ':';

    return (
      <fieldset className='account-input' form={this.props.form} name={this.props.id}>
        <div className='main-input'>
          <label
            htmlFor={this.props.id}
            className='input-label'
          > {inputLabel}
            <span className='hint'>{this.props.hint}</span>
          </label>
          <input
            type={this.props.type}
            id={this.props.id}
            name={this.props.id}
            ref={this.props.inputRef || null}
            aria-required='true'
            aria-describedby={this.props.id + '-error-message'}
            aria-invalid={!!this.props.validate()}
            onChange={(e) => this.props.update(e.target.value)} />
        </div>
        {this.props.touched && <ValidationError id={this.props.id + '-error-message'} errorFor={this.props.id} message={this.props.validate()} />}
      </fieldset>
    );
  };
}

AccountInput.defaultProps = {
  form: '',
  id: '',
  type: '',
  touched: false,
  validate: () => { },
  update: () => { },
  inputRef: null,
  hint: ''
}

AccountInput.propTypes = {
  form: PropTypes.string.isRequired,
  id: PropTypes.string,
  type: PropTypes.string.isRequired,
  touched: PropTypes.bool.isRequired,
  validate: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  inputRef: PropTypes.func,
  hint: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string
  ])
}

export default AccountInput;