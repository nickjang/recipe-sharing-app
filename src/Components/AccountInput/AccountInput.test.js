import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AccountInput from './AccountInput';

describe(`AccountInput component`, () => {
  it('renders AccountInput by default', () => {
    const wrapper = shallow(<AccountInput />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})