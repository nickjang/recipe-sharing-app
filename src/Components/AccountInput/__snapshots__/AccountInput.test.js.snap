// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInput component renders AccountInput by default 1`] = `
<fieldset
  className="account-input"
  form=""
  name=""
>
  <div
    className="main-input"
  >
    <label
      className="input-label"
      htmlFor=""
    >
       
      :
      <span
        className="hint"
      />
    </label>
    <input
      aria-describedby="-error-message"
      aria-invalid={false}
      aria-required="true"
      id=""
      name=""
      onChange={[Function]}
      type=""
    />
  </div>
</fieldset>
`;
